azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualMachineScaleSetOSProfile.java

```diff
@@ -0,0 +1,267 @@
+      public static WindowsConfiguration create(final boolean provisionVMAgent, final WinRM winRM,
+                                                final List<AdditionalUnattendContent> additionalUnattendContent,
+                                                final boolean enableAutomaticUpdates) {
+
+         return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration(provisionVMAgent, winRM,
+                 additionalUnattendContent, enableAutomaticUpdates);
+      }
+   }
+
+   /**
+    * The computer name of the VM
+    */
+   @Nullable
+   public abstract String computerNamePrefix();
+
+   /**
+    * The admin username of the VM
+    */
+   @Nullable
+   public abstract String adminUsername();
+
+   /**
+    * The admin password of the VM
+    */
+   @Nullable
+   public abstract String adminPassword();
+
+   /**
+    * The linux configuration of the VM
+    */
+   @Nullable
+   public abstract LinuxConfiguration linuxConfiguration();
+
+   /**
+    * The windows configuration of the VM
+    */
+   @Nullable
+   public abstract WindowsConfiguration windowsConfiguration();
+
+   /**
+    * The Secrets configuration of the VM
+    */
+   @Nullable
+   public abstract List<Secrets> secrets();
+
+   @SerializedNames({"computerNamePrefix", "adminUsername", "adminPassword", "linuxConfiguration",
+           "windowsConfiguration", "secrets"})
+   public static VirtualMachineScaleSetOSProfile create(final String computerNamePrefix, final String adminUsername,
+                                                        final String adminPassword, final LinuxConfiguration linuxConfiguration,
+                                                        final WindowsConfiguration windowsConfiguration, final List<Secrets> secrets) {
+      return builder()
+              .computerNamePrefix(computerNamePrefix)
+              .adminUsername(adminUsername)
+              .adminPassword(adminPassword)
+              .linuxConfiguration(linuxConfiguration)
+              .windowsConfiguration(windowsConfiguration)
+              .secrets(secrets)
+              .build();
+   }
+   
+   public abstract Builder toBuilder();
+
+   public static Builder builder() {
+      return new AutoValue_VirtualMachineScaleSetOSProfile.Builder();
+   }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+      public abstract Builder computerNamePrefix(String computerNamePrefix);
+
+      public abstract Builder adminUsername(String adminUsername);
+
+      public abstract Builder adminPassword(String adminPassword);
+
+      public abstract Builder linuxConfiguration(LinuxConfiguration linuxConfiguration);
+
+      public abstract Builder windowsConfiguration(WindowsConfiguration windowsConfiguration);
+
+      public abstract Builder secrets(List<Secrets> secrets);
+
+      public abstract VirtualMachineScaleSetOSProfile build();
+   }
+}
```

Comment:
is this class really different from the existing `OSProfile`? If it really is, extract the common classes from the existing one so they can be reused.